src/test/java/se/sundsvall/messaging/api/MessageResourceEmailBatchTests.java

```diff
@@ -0,0 +1,111 @@
+package se.sundsvall.messaging.api;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+import static org.springframework.http.HttpHeaders.LOCATION;
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static se.sundsvall.messaging.TestDataFactory.createValidEmailBatchRequest;
+import static se.sundsvall.messaging.model.MessageStatus.SENT;
+import static se.sundsvall.messaging.model.MessageType.EMAIL;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.reactive.server.WebTestClient;
+
+import se.sundsvall.messaging.Application;
+import se.sundsvall.messaging.api.model.response.MessageBatchResult;
+import se.sundsvall.messaging.model.InternalDeliveryBatchResult;
+import se.sundsvall.messaging.model.InternalDeliveryResult;
+import se.sundsvall.messaging.service.MessageEventDispatcher;
+import se.sundsvall.messaging.service.MessageService;
+
+@SpringBootTest(classes = Application.class, webEnvironment = RANDOM_PORT)
+@ActiveProfiles("junit")
+class MessageResourceEmailBatchTests {
+
+	private static final String URL = "/email/batch";
+	private static final String ORIGIN_HEADER = "x-origin";
+	private static final String ORIGIN = "origin";
+	private static final InternalDeliveryResult DELIVERY_RESULT = InternalDeliveryResult.builder()
+		.withMessageId("someMessageId")
+		.withDeliveryId("someDeliveryId")
+		.withMessageType(EMAIL)
+		.withStatus(SENT)
+		.build();
+	private static final InternalDeliveryBatchResult DELIVERY_BATCH_RESULT = InternalDeliveryBatchResult.builder()
+		.withBatchId("someBatchId")
+		.withDeliveries(List.of(DELIVERY_RESULT))
+		.build();
+
+	@MockBean
+	private MessageService mockMessageService;
+
+	@MockBean
+	private MessageEventDispatcher mockEventDispatcher;
+
+	@Autowired
+	private WebTestClient webTestClient;
+
+	@ParameterizedTest
+	@MethodSource("requestProvider")
+	void sendBatch(String senderName, String partyId) {
```

Comment:
Parameters are not used?

Reply:
Will remove, they serve no purpose in this test